internal/storage/disk/dirwatch.go

```diff
@@ -110,18 +110,19 @@ func (dw *dirWatch) processEvent(evtInfo notify.EventInfo) {
 
 func (dw *dirWatch) triggerUpdate() {
 	dw.mu.RLock()
-	shouldUpdate := len(dw.eventBatch) > 0 && (time.Since(dw.lastEventTime) > dw.cooldownPeriod)
+	shouldUpdate := len(dw.eventBatch) > 0 && (time.Since(dw.lastEventTime).Microseconds() > dw.cooldownPeriod.Microseconds())
 	dw.mu.RUnlock()
 
 	//nolint:nestif
 	if shouldUpdate {
 		dw.mu.Lock()
-		proceed := len(dw.eventBatch) > 0 && (time.Since(dw.lastEventTime) > dw.cooldownPeriod)
+		proceed := len(dw.eventBatch) > 0 && (time.Since(dw.lastEventTime).Microseconds() > dw.cooldownPeriod.Microseconds())
 		if !proceed {
 			dw.mu.Unlock()
 			return
 		}
 
+		ts := dw.lastEventTime
```

Comment:
I think the `ts` should be `time.Now()` because the actual store refresh happened just now.